### PR TITLE
Make new functions adapted for JSON RPC v2.0

### PR DESCRIPTION
**Pull request by boyeon555**

### CODE REVIEW

File: icx/utils.py [hunk 1]
```diff
@@ -395,6 +395,61 @@ def get_balance(address, url):
...
+    payload = create_jsonrpc_request_content(0, method, params)
+    response = post(url, payload)
+    content = response.json()
+    package_json_contents = json.dumps(content, indent=4)
+
+    return package_json_contents
+
+
+def get_block_by_height(height, url):
+    """ Get block information by height.
```
Comment (JinhoYoo-icon-project) [inline icx/utils.py]: No explanation about parameters and return value in docstring.

File: icx/utils.py [hunk 2]
```diff
@@ -395,6 +395,61 @@ def get_balance(address, url):
...
+    payload = create_jsonrpc_request_content(0, method, params)
+    response = post(url, payload)
+    content = response.json()
+    package_json_contents = json.dumps(content, indent=4)
+
+    return package_json_contents
+
+
+def get_last_block(url):
+    """ Get last block information.
```
Comment (JinhoYoo-icon-project) [inline icx/utils.py]: No explanation about parameters and return value in docstring.

File: icx/wallet/wallet.py [hunk 3]
```diff
@@ -220,4 +221,35 @@ def get_address(self):
         """
         return self.address
 
+    @staticmethod
```
Comment (JinhoYoo-icon-project) [inline icx/wallet/wallet.py]: Use @classmethod, not @staticmethod. See [this](https://github.com/shoark7/Effective-Python/blob/master/files/BetterWay24_classmethod.md) document.

File: icx/wallet/wallet.py [hunk 4]
```diff
@@ -220,4 +221,35 @@ def get_address(self):
...
+        """ get block information by height
+
+        :param height:
+        :param uri type(str)
+        :return:
+        """
+        block = get_block_by_height(height, uri)
+        return block
+
+    @staticmethod
```
Comment (JinhoYoo-icon-project) [inline icx/wallet/wallet.py]: Use @classmethod, not @staticmethod. See [this](https://github.com/shoark7/Effective-Python/blob/master/files/BetterWay24_classmethod.md) document.

File: icx/wallet/wallet.py [hunk 5]
```diff
@@ -220,4 +221,35 @@ def get_address(self):
...
+        """ get block information by hash
+
+        :param hash:
+        :param uri:
+        :return:
+        """
+        block = get_block_by_hash(hash, uri)
+        return block
+
+    @staticmethod
```
Comment (JinhoYoo-icon-project) [inline icx/wallet/wallet.py]: Use @classmethod, not @staticmethod. See [this](https://github.com/shoark7/Effective-Python/blob/master/files/BetterWay24_classmethod.md) document.

File: tests/test_get_block_by_hash.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+class TestGetBlockByHash(unittest.TestCase):
+
+    def test0(self):
+        """ Case to get block by hash successfully
+        """
+        # Given, When
+        block = Wallet.get_block_by_hash('885b8021826f7e741be7f53bb95b48221e9ab263f377e997b2e47a7b8f4a2a8b')
+
+        # Then
+        self.assertEqual(0, json.loads(block)["result"]['response_code'])
```
Comment (JinhoYoo-icon-project) [inline tests/test_get_block_by_hash.py]: Why didn't you check the fields of return data? Please check the field of return values. See [this](https://github.com/icon-project/icx_JSON_RPC).

File: tests/test_get_block_by_height.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+class TestGetBlockByHeight(unittest.TestCase):
+
+    def test0(self):
+        """ Case to get block by Height successfully
+        """
+        # Given, When
+        block = Wallet.get_block_by_height(1)
+
+        # Then
+        self.assertEqual(0, json.loads(block)["result"]['response_code'])
```
Comment (JinhoYoo-icon-project) [inline tests/test_get_block_by_height.py]: Why didn't you check the fields of return data? Please check the field of return values. See [this](https://github.com/icon-project/icx_JSON_RPC).

File: tests/test_get_last_block.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+class TestGetLastBlock(unittest.TestCase):
+
+    def test0(self):
+        """ Case to get last block successfully
+        """
+        # Given, When
+        block = Wallet.get_last_block()
+
+        # Then
+        self.assertEqual(0, json.loads(block)["result"]['response_code'])
```
Comment (JinhoYoo-icon-project) [inline tests/test_get_last_block.py]: Why didn't you check the fields of return data? Please check the field of return values. See [this](https://github.com/icon-project/icx_JSON_RPC).

File: icx/utils.py [hunk 9]
```diff
@@ -395,6 +395,61 @@ def get_balance(address, url):
     return dec_loop_balance
 
 
+def get_block_by_hash(hash, url):
+    """ Get block information by hash.
+
+    :param hash: Using hash values ​​with electronic signatures. 64 character. hexadecimal.
+    :param url:
```
[the remark this answers is not in the document]
Comment (boyeon555) [reply icx/utils.py]: Added it.

File: icx/utils.py [hunk 10]
```diff
@@ -395,6 +395,61 @@ def get_balance(address, url):
...
+    :return:
```
[the remark this answers is not in the document]
Comment (boyeon555) [reply icx/utils.py]: Added it.

File: icx/utils.py (hunk 1 above)
Comment (boyeon555) [reply icx/utils.py]: Added it.

File: icx/utils.py (hunk 2 above)
Comment (boyeon555) [reply icx/utils.py]: Added it.

File: icx/wallet/wallet.py (hunk 3 above)
Comment (boyeon555) [reply icx/wallet/wallet.py]: Fixed it.

File: icx/wallet/wallet.py (hunk 4 above)
Comment (boyeon555) [reply icx/wallet/wallet.py]: Fixed it.

File: icx/wallet/wallet.py (hunk 5 above)
Comment (boyeon555) [reply icx/wallet/wallet.py]: Fixed it.

File: tests/test_get_block_by_hash.py (hunk 6 above)
Comment (boyeon555) [reply tests/test_get_block_by_hash.py]: Added it.

File: tests/test_get_block_by_height.py (hunk 7 above)
Comment (boyeon555) [reply tests/test_get_block_by_height.py]: Added it.

File: tests/test_get_last_block.py (hunk 8 above)
Comment (boyeon555) [reply tests/test_get_last_block.py]: Added it.

**Comment by JinhoYoo-icon-project** [comment]
Good job!

**Comment by JinhoYoo-icon-project** [comment]
Thank you for your contribution. ;-)

**Pull request closed without merging**